### PR TITLE
Switch to a single, released version of the Prismic libraries

### PR DESCRIPTION
**Pull request by alexwlchan**
This gets our main web apps onto a single version of the Prismic client libraries, using the migration guides published by Prismic:

* \@prismicio/client: https://prismic.io/docs/technologies/prismic-client-v6-migration-guide
* \@prismicio/helpers: https://prismic.io/docs/technologies/prismic-helpers-v2-migration-guide
* \@prismicio/react: https://prismic.io/docs/technologies/prismic-react-v2-migration-guide

They claim the new libraries have smaller bundle sizes, so 🤞 for noticeable savings. The old `prismic-react` library is certainly a big part of our current bundle.

Mostly the changes are pretty simple – the HTML serialiser is the only component with any non-trivial changes. I had to add a few extra type hints to help the type checker, but otherwise it was pretty straightforward.

We're still using the old Prismic libraries in the RSS app, but refactoring that is a non-trivial task that we should tackle separately. See https://github.com/wellcomecollection/wellcomecollection.org/issues/7751 for details.

I'm running it through the bulk page tester to look for errors, and I'll mark this as non-WIP when I've fixed any bugs that throws up.

Closes https://github.com/wellcomecollection/wellcomecollection.org/issues/7595

### CODE REVIEW

File: common/views/components/ButtonInline/ButtonInline.tsx [hunk 1]
```diff
@@ -93,6 +93,4 @@ const ButtonInline = forwardRef<HTMLButtonElement, ButtonInlineProps>(
   }
 );
 
-ButtonInline.displayName = 'ButtonInline';
```
Comment (davidpmccormick) [inline common/views/components/ButtonInline/ButtonInline.tsx:96]: Screenshot 2022-03-07 at 16 17 57

Hands up! It's the lint police.

**Review by melanierogan — APPROVED** [review]
I've not made any specifically helpful comments on lines of code, it looks good to me. I dropped a few non-blocking questions. 
Nice work!
 an-animated-gif-of-adventuretime-character-giving-thumbs-up

File: content/webapp/services/prismic/transformers/images.ts [hunk 2]
```diff
@@ -49,10 +50,14 @@ export function transformCaptionedImage(
 export function transformPromoToCaptionedImage(
   frag: PromoSliceZone,
   crop?: Crop
-): CaptionedImage {
+): CaptionedImage | undefined {
   // We could do more complicated checking here, but this is what we always use.
```
[the remark this answers is not in the document]
Comment (alexwlchan) [reply content/webapp/services/prismic/transformers/images.ts:54]: Good question! I didn't write this comment originally, so I'm not sure 😅 

I would guess it's around getting the first element of `frag` and assuming there's only one; what we do if e.g. there aren't any?

File: content/webapp/services/prismic/types/events.ts [hunk 3]
```diff
@@ -167,7 +169,13 @@ export type EventPrismicDocument = PrismicDocument<
       event: RelationField<
         'events',
         'en-gb',
-        InferDataInterface<EventPrismicDocument>
+        // We know this is an EventPrismicDocument, but the type checker gets
```
[the remark this answers is not in the document]
Comment (alexwlchan) [reply content/webapp/services/prismic/types/events.ts:172]: I'm not going to say "no", but we haven't typically made tickets for TODO comments like this in the past – they tend to get picked up as part of "if you're working on this, leave it in a better state" type work.

But if you'd find it helpful or you think you might return to this later, don't let me stop you!

**Pull request merged**